src/Framework/Sdk/SdkResultItem.cs

```diff
@@ -36,33 +41,21 @@ public override bool Equals(object obj)
                    ItemSpec == item.ItemSpec &&
                    Metadata?.Count == item.Metadata?.Count)
             {
-                if (Metadata != null)
-                {
-                    foreach (var kvp in Metadata)
-                    {
-                        if (item.Metadata[kvp.Key] != kvp.Value)
-                        {
-                            return false;
-                        }
-                    }
-                }
-
-                return true;
+                return Metadata.All(m => item.Metadata.TryGetValue(m.Key, out var itemValue) && itemValue == m.Value);
```

Comment:
```suggestion
                return Metadata.All(m => item.Metadata?.TryGetValue(m.Key, out var itemValue) && itemValue == m.Value);
```

Reply:
@Forgind 

if I apply your suggestion, I get this error compile"
``` 
Operator '&&' cannot be applied to operands of type 'bool?' and 'bool'
```

Because I think this part `item.Metadata?.TryGetValue(m.Key, out var itemValue)` is evaluated to bool? instead of bool.

Please elaborate or confirm...

Reply:
@Forgind 

Please review the failing CI build too, if you don't mind..
I have no clue which failing tests causing the error/failures.

Reply:
Ah, that's true. You can check if the TryGetValue equals true, and that should resolve it.

Reply:
Thanks! resolving this now.